src/Integration.UnitTests/SanityTests.cs

```diff
@@ -49,40 +52,42 @@ public void LatestServer_APICompatibility()
         {
             // Setup the service used to interact with SQ
             var s = new SonarQubeServiceWrapper(this.serviceProvider);
+            var connection = new ConnectionInformation(new Uri("http://nemo.sonarqube.org"));
 
             // Step 1: Connect anonymously
-            var projects = RetryAction(() => s.Connect(new ConnectionInformation(new Uri("http://nemo.sonarqube.org")), CancellationToken.None),
+            ProjectInformation[] projects = null;
+
+            RetryAction(() => s.TryGetProjects(connection, CancellationToken.None, out projects),
                                         "Get projects from SonarQube server");
-            Assert.AreNotEqual(0, projects.Count(), "No projects were returned");
+            Assert.AreNotEqual(0, projects.Length, "No projects were returned");
 
             // Step 2: Get quality profile export for the first project
             var project = projects.FirstOrDefault();
-            var export = RetryAction(() => s.GetExportProfile(project, SonarQubeServiceWrapper.CSharpLanguage, CancellationToken.None),
+            RoslynExportProfile export = null;
+            RetryAction(() => s.TryGetExportProfile(connection, project, SonarQubeServiceWrapper.CSharpLanguage, CancellationToken.None, out export),
                                         "Get quality profile export from SonarQube server");
             Assert.IsNotNull(export, "No quality profile export was returned");
 
             // Errors are logged to output window pane and we don't expect any
             this.outputWindowPane.AssertOutputStrings(0);
         }
 
-        private static T RetryAction<T>(Func<T> action, string description, int maxAttempts = 3)
-            where T: class
+        private static bool RetryAction(Func<bool> action, string description, int maxAttempts = 3)
         {
             for (int attempt = 0; attempt < maxAttempts; attempt++)
             {
-                T value = action();
-                if (value == null)
+                if (!action())
                 {
                     Thread.Sleep(100);
                 }
                 else
                 { 
-                    return value;
+                    return true;
```

Comment:
Why not just return nothing (`void`) since in the failure case the test will be stopped/failed to the `Assert.Fail`.